### PR TITLE
chore: now docs are autogenerated as part of the release process

### LINKED ISSUE #152: Feature: automate docs generation in release action
### Suggested solution

Now that #143 has landed to enable the `pizza docs` command, we can use this in the release action to automate generating and committing new docs during the release action.

In https://github.com/open-sauced/pizza-cli/pull/151, I included the `just gen-docs` command which will execute `go run main.go docs` to generate the docs. We can use this with the existing `just` and `go` setup to _also_ generate and commit the docs during the release action here:

https://github.com/open-sauced/pizza-cli/blob/3cdf5ceed8d0a273a976fdcd6e0ed94e94cab265/.github/workflows/release.yaml#L80-L86

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

This pull request introduces a new job to the release workflow for automating the documentation update process before releasing a new version. Additionally, it ensures that the release process waits for the documentation update PR to be merged before proceeding.

The repository is already configured for GitHub Actions to create and approve PRs.

CleanShot 2024-09-06 at 14 43 34

## Related Tickets & Documents

Closes #152

## Steps to QA

I've tested my script, but not the whole flow. We can see how it behaves for a beta release.

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

File: .github/workflows/release.yaml
Comment (nickytonline) [inline .github/workflows/release.yaml:67]: This seems like a good approach, but would love feedback. Adding it as a step without a `needs` means the PR might not be merged for the docs yet before a release happens, and it can also cause an infinite loop of a release.

I put tests in here as well.

File: .github/workflows/release.yaml
Comment (nickytonline) [inline .github/workflows/release.yaml:113]: There is a risk of this never finishing, so we could add a retry max in here.

File: .github/workflows/release.yaml
Comment (nickytonline) [inline .github/workflows/release.yaml:98]: We skip the `"Wait for PR to be merged"` if no PR was created, i.e. up top

```
if git diff --staged --quiet; then
  echo "No changes to commit"
  echo "pr_number=" >> $GITHUB_OUTPUT
```

File: .github/workflows/release.yaml
Comment (nickytonline) [inline .github/workflows/release.yaml:49]: These could be a script but I opted for inlining for simplicity.

File: .github/workflows/release.yaml
Comment (jpmcb) [reply .github/workflows/release.yaml:67]: I'd honestly go even simpler: we already automatically commit the `CHANGELOG.md` in the release action. So I'd say this could just be an additional step before the semantic-release flow (although, I think semantic-release fails if the git repo is dirty so it'd probably have to be a separate, quick followup commit).

So instead of creating a whole PR, I'd just commit it directly to the target release branch with `skip-ci` like we do with other release assets.

Although I could see the value in getting eyes on the generated docs via PR before merging them in. It could be easy to let the process get away from us and we never actually look at what's being generated.

File: .github/workflows/release.yaml
Comment (jpmcb) [inline .github/workflows/release.yaml:42]: This is missing a "setup go" step:

```yaml
    - name: "🐹 Setup Go"
      uses: actions/setup-go@v5
      with:
        go-version: 1.22.x
```

I don't _think_ the GitHub runners have Go installed, but before running `just gen-docs` (which executes `go run`), I'd go ahead and get Go setup

File: .github/workflows/release.yaml
Comment (nickytonline) [reply .github/workflows/release.yaml:67]: I wasn't sure if we wanted to commit and push, so sounds good!

**Review by jpmcb — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**
